src/main/java/com/ibdiscord/listeners/ButtonListener.java

```diff
@@ -60,10 +60,11 @@ public void onButtonClick(@NotNull ButtonClickEvent event) {
                 toAdd.add(role);
             }
         }
-        guild.modifyMemberRoles(member, toAdd, new ArrayList<>()).queue(yes -> {
-            event.reply(generateMessage(toAdd.size(), 0))
-                .setEphemeral(true)
-                .queue();
+        event.deferReply(true).queue(deferred -> {
+                guild.modifyMemberRoles(member, toAdd, new ArrayList<>()).queue(yes -> {
+                    event.getHook().sendMessage(generateMessage(toAdd.size(), 0))
```

Comment:
Does this stop the loading... animation when it sends the message after a while? What happens if the `sendMessage` gets called like 10 seconds later?

Reply:
I've tested artificial delays (with Thread.sleep) and it works as intended (e.g., continues loading animation and then edits the message after action completes).